pkg/sensors/tracing/genericlsm.go

```diff
@@ -342,23 +371,99 @@ func createGenericLsmSensor(
 	}, nil
 }
 
+func imaProgName(lsmEntry *genericLsm) (string, string) {
+	pType := ""
+	pName := ""
+
+	switch lsmEntry.hook {
+	case "bprm_check_security":
+		fallthrough
+	case "bprm_committed_creds":
+		fallthrough
+	case "bprm_committing_creds":
+		fallthrough
+	case "bprm_creds_for_exec":
+		fallthrough
+	case "bprm_creds_from_file":
+		pType = "bprm"
+	case "file_ioctl":
+		fallthrough
+	case "file_lock":
+		fallthrough
+	case "file_open":
+		fallthrough
+	case "file_post_open":
+		fallthrough
+	case "file_receive":
+		fallthrough
+	case "mmap_file":
+		pType = "file"
+	default:
+		return "", ""
+	}
+	if kernels.EnableV61Progs() {
+		pName = "bpf_generic_lsm_ima_" + pType + "_v61.o"
+	} else if kernels.MinKernelVersion("5.11") {
+		pName = "bpf_generic_lsm_ima_" + pType + "_v511.o"
+	}
+	return pName, pType
+}
+
 func createLsmSensorFromEntry(lsmEntry *genericLsm,
 	progs []*program.Program, maps []*program.Map) ([]*program.Program, []*program.Map) {
 
-	loadProgName := "bpf_generic_lsm.o"
+	loadProgCoreName := "bpf_generic_lsm_core.o"
+	loadProgOutputName := "bpf_generic_lsm_output.o"
 	if kernels.EnableV61Progs() {
-		loadProgName = "bpf_generic_lsm_v61.o"
+		loadProgCoreName = "bpf_generic_lsm_core_v61.o"
+		loadProgOutputName = "bpf_generic_lsm_output_v61.o"
 	} else if kernels.MinKernelVersion("5.11") {
-		loadProgName = "bpf_generic_lsm_v511.o"
+		loadProgCoreName = "bpf_generic_lsm_core_v511.o"
+		loadProgOutputName = "bpf_generic_lsm_output_v511.o"
 	}
 
+	loadOutput := program.Builder(
+		path.Join(option.Config.HubbleLib, loadProgOutputName),
+		lsmEntry.hook,
+		"lsm/generic_lsm_output",
+		lsmEntry.hook,
+		"generic_lsm").
+		SetLoaderData(lsmEntry.tableId)
+	progs = append(progs, loadOutput)
+
 	load := program.Builder(
-		path.Join(option.Config.HubbleLib, loadProgName),
+		path.Join(option.Config.HubbleLib, loadProgCoreName),
 		lsmEntry.hook,
-		"lsm/generic_lsm",
+		"lsm/generic_lsm_core",
 		lsmEntry.hook,
 		"generic_lsm").
 		SetLoaderData(lsmEntry.tableId)
+
+	// Load ima program for hash calculating
```

Comment:
could you please make some note in here that this is the part (I hope it is) where we set the order of the trampoline programs and that we might do feature detection for the programs order in the trampoline? thnx